crates/orchestrator/src/routes/job_routes.rs

```diff
@@ -73,6 +74,65 @@ async fn handle_verify_job_request(
         }
     }
 }
+
+async fn handle_retry_job_request(
+    Path(JobId { id }): Path<JobId>,
+    State(config): State<Arc<Config>>,
+) -> impl IntoResponse {
+    let job_id = match Uuid::parse_str(&id) {
+        Ok(id) => id,
+        Err(_) => {
+            return ApiResponse::<JobApiResponse>::error((JobError::InvalidId { id }).to_string()).into_response();
+        }
+    };
+
+    // Get the job and verify it's in a failed state
+    let job = match config.database().get_job_by_id(job_id).await {
+        Ok(Some(job)) => job,
+        Ok(None) => {
+            return ApiResponse::<JobApiResponse>::error(JobError::JobNotFound { id: job_id }.to_string())
+                .into_response();
+        }
+        Err(e) => {
+            return ApiResponse::<JobApiResponse>::error(e.to_string()).into_response();
+        }
+    };
+
+    // Check if job is in a failed state
+    if job.status != JobStatus::Failed {
+        return ApiResponse::<JobApiResponse>::error(format!(
+            "Job {} cannot be retried: current status is {:?}",
+            id, job.status
+        ))
+        .into_response();
+    }
+
+    // Update the job status to RetryAttempt
+    match config.database().update_job(&job, JobItemUpdates::new().update_status(JobStatus::RetryAttempt).build()).await
+    {
+        Ok(_) => {
+            // Process the job after successful status update
+            match process_job(job_id, config).await {
+                Ok(_) => {
+                    let response =
+                        JobApiResponse { job_id: job_id.to_string(), status: "retry_processing".to_string() };
+                    ApiResponse::success(response).into_response()
+                }
+                Err(e) => {
+                    ORCHESTRATOR_METRICS
+                        .failed_job_operations
+                        .add(1.0, &[KeyValue::new("operation_type", "retry_job")]);
```

Comment:
operation_type should strictly be `process_job` since that's the function being called.
you can add another attribute  like `operation_info` for registering `retry_job`